src/source/azuread.ts

```diff
@@ -87,31 +89,42 @@ class AzureAD {
       return appInfoKeys.every(key => Object.prototype.hasOwnProperty.call(obj, key))
     }
     // PromisePoolで並列処理を実行
-    const { results } = await PromisePool
-      .for<AppInfo>(apps.filter(isAppInfo)) // 型ガードでフィルタリング
-      .withConcurrency(5) // 並列数を5に制限
-      .process(async (app: AppInfo) => {
-        // Get the service principal id
-        const servicePrincipalUrl = `https://graph.microsoft.com/v1.0/servicePrincipals?$filter=appId eq '${app.appId}'`
-        const servicePrincipalResponse = await axios.get(servicePrincipalUrl, {
-          headers: {
-            Authorization: `Bearer ${accessToken}`
+    let appInfos = [] as AppInfo[]
+    try {
+      const { results, errors } = await PromisePool
+        .for<AppInfo>(apps.filter(isAppInfo)) // 型ガードでフィルタリング
+        .withConcurrency(5) // 並列数を5に制限
+        .process(async (app: AppInfo) => {
+          // Get the service principal id
+          const servicePrincipalUrl = new URL('https://graph.microsoft.com/v1.0/servicePrincipals');
+          servicePrincipalUrl.searchParams.set('$filter', `appId eq '${app.appId}'`)
+          const servicePrincipalResponse = await axios.get(servicePrincipalUrl.toString(), {
+            headers: {
+              Authorization: `Bearer ${accessToken}`
+            }
+          })
+          const servicePrincipalId = servicePrincipalResponse.data.value[0]?.id
+          const tags: string[] = servicePrincipalResponse.data.value[0]?.tags ?? []
+          const appInfo: AppInfo = {
+            appId: app.appId,
+            displayName: app.displayName,
+            principleId: servicePrincipalId,
+            signInAudience: app.signInAudience,
+            identifierUris: app.identifierUris,
+            tags: tags,
+            users: []
           }
+          return appInfo
         })
-        const servicePrincipalId = servicePrincipalResponse.data.value[0]?.id
-        const tags: string[] = servicePrincipalResponse.data.value[0]?.tags ?? []
-        const appInfo: AppInfo = {
-          appId: app.appId,
-          displayName: app.displayName,
-          principleId: servicePrincipalId,
-          signInAudience: app.signInAudience,
-          identifierUris: app.identifierUris,
-          tags: tags,
-          users: []
+      errors.forEach(error => {
+        if (error instanceof Error) {
+          throw new Error(error.message);
         }
-        return appInfo
       })
-    let appInfos = results
+      appInfos = results
+    } catch (error) {
+      throw new Error(`Fail to fetch applications from AzureAd: ${error}`);
+    }
```

Comment:
same as in other place, try-catch not needed and print messages and do fnal throw